### PR TITLE
Add the CI test by github action

### PR DESCRIPTION
**Pull request by bCoder778**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -26,3 +26,28 @@ jobs:
 
     - name: Build
       run: go build -v .
+      
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.13
```
Comment (dindinw) [inline .github/workflows/go.yml]: why an older version? any necessity here?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -26,3 +26,28 @@ jobs:
...
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
+
+    - name: Test
+      run: go test -v ./testutils
```
Comment (dindinw) [inline .github/workflows/go.yml]: why specify package instead of all?  `go test -v ./...`

File: testutils/wallet.go [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package testutils
+
+func CreateWallet(){
+
```
Comment (dindinw) [inline testutils/wallet.go]: why 2 empty tests?

File: .github/workflows/go.yml (hunk 1 above)
Comment (bCoder778) [reply .github/workflows/go.yml]: OK, I'll fix it.

File: testutils/wallet.go (hunk 3 above)
Comment (bCoder778) [reply testutils/wallet.go]: Testutils is temporarily on the 0.10-dev branch, but in order not to let Matser report an error, there is only an empty test.

File: .github/workflows/go.yml (hunk 2 above)
Comment (bCoder778) [reply .github/workflows/go.yml]: ok

File: internal/legacy/keystore/keystore_test.go [hunk 4]
```diff
@@ -172,34 +172,34 @@ func TestChaining(t *testing.T) {
 			cc:             []byte("3318959fff419ab8b556facb3c429a86"),
 			origPrivateKey: []byte("5ffc975976eaaa1f7b179f384ebbc053"),
 			nextPrivateKeyUncompressed: []byte{
-				0xd3, 0xfe, 0x2e, 0x96, 0x44, 0x12, 0x2d, 0xaa,
-				0x80, 0x8e, 0x36, 0x17, 0xb5, 0x9f, 0x8c, 0xd2,
-				0x72, 0x8c, 0xaf, 0xf1, 0xdb, 0xd6, 0x4a, 0x92,
-				0xd7, 0xc7, 0xee, 0x2b, 0x56, 0x34, 0xe2, 0x87,
+				0x70 ,0x7b ,0x46 ,0x99 ,0x09 ,0x09 ,0x4f ,0x97,
```
Comment (dindinw) [inline internal/legacy/keystore/keystore_test.go:175]: why we need to change priv/pub/sign etc related code to fix the unit-test?  I am not sure why it required for the master branch?

**Comment by dindinw** [comment]
I think you need to upgrade the current action template to a newer version, instead keep using the old one (17 months+ old). you can build a quick template  by by clicking `Actions` button in any your own project and select `go` language.

```
name: Go

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Set up Go
      uses: actions/setup-go@v2
      with:
        go-version: 1.15

    - name: Build
      run: go build -v ./...

    - name: Test
      run: go test -v ./...
```
see more https://github.com/actions/starter-workflows/blob/main/ci/go.yml

**Pull request merged**